craft_providers/multipass/multipass.py

```diff
@@ -0,0 +1,381 @@
+# Copyright 2021 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""API provider for Multipass.
+
+This implementation interfaces with multipass using the `multipass` command-line
+utility.
+"""
+
+import io
+import json
+import logging
+import pathlib
+import shlex
+import subprocess
+from typing import Any, Dict, List, Optional
+
+from craft_providers import errors
+
+from .errors import MultipassError
```

Comment:
Yes isort groups relative imports separately by default.  We can add `no_lines_before = ["LOCALFOLDER"]` to the config to prevent that, if you'd like.  I like the separate grouping, but don't mind either way.

Reply:
this is fine, I am fine with automation :-)